sql/table.go

```diff
@@ -365,8 +393,35 @@ func convertDatum(col ColumnDescriptor, val parser.Datum) (interface{}, error) {
 			return float64(v), nil
 		}
 	// case ColumnType_DECIMAL:
-	// case ColumnType_DATE:
-	// case ColumnType_TIMESTAMP:
+	case ColumnType_DATE:
+		switch val := val.(type) {
+		case parser.DString:
```

Comment:
this looks a sketchy; why are we parsing in this method? at least, this deserves a comment since this is different from other handling here